sdk/storage/storage-file-share/test/directoryclient.spec.ts

```diff
@@ -2464,3 +2471,132 @@ describe("DirectoryClient - AllowingTrailingDots - Default", () => {
     assert.equal(await dirClient1.exists(), false);
   });
 });
+
+describe("DirectoryClient - NFS", () => {
+  let recorder: Recorder;
+  let shareName: string;
+  let shareClient: ShareClient;
+  let dirName: string;
+  let dirClient: ShareDirectoryClient;
+
+  beforeEach(async function (this: Context) {
+    recorder = new Recorder(this.currentTest);
+    await recorder.start(recorderEnvSetup);
+    await recorder.addSanitizers({ uriSanitizers }, ["record", "playback"]);
+    let serviceClient: ShareServiceClient;
+    try {
+      serviceClient = getGenericBSU(recorder, "PREMIUM_FILE_");
+    } catch (error: any) {
+      console.log(error);
+      this.skip();
+    }
+
+    shareName = recorder.variable("share", getUniqueName("share"));
+    shareClient = serviceClient.getShareClient(shareName);
+    await shareClient.create({
+      protocols: {
+        smbEnabled: true,
+        nfsEnabled: true,
+      },
+    });
+
+    dirName = recorder.variable("dir", getUniqueName("dir"));
+    dirClient = shareClient.getDirectoryClient(dirName);
+  });
+
+  afterEach(async function (this: Context) {
+    if (shareClient) {
+      await shareClient.delete({ deleteSnapshots: "include" });
+    }
+    await recorder.stop();
+  });
+
+  it("create with nfs properties", async function () {
+    const posixProperties: FilePosixProperties = {
+      owner: "123",
+      group: "654",
+      fileMode: {
+        owner: {
+          read: true,
+          write: true,
+          execute: true,
+        },
+        group: {
+          read: true,
+          write: false,
+          execute: true,
+        },
+        other: {
+          read: true,
+          write: false,
+          execute: true,
+        },
+        effectiveUserIdentity: false,
+        effectiveGroupIdentity: false,
+        stickyBit: false,
+      },
+    };
+    const cResp = await dirClient.create({
+      posixProperties: posixProperties,
+    });
+
+    assert.equal(cResp.errorCode, undefined);
+    assert.deepEqual(cResp.owner, posixProperties.owner);
+    assert.deepEqual(cResp.group, posixProperties.group);
+    assert.deepEqual(cResp.fileMode, posixProperties.fileMode);
+    assert.deepEqual(cResp.nfsFileType, "Directory");
+    assert.ok(cResp.fileChangeOn!);
+    assert.ok(cResp.fileCreatedOn!);
+    assert.ok(cResp.fileId!);
+    assert.ok(cResp.fileLastWriteOn!);
+    assert.ok(cResp.fileParentId!);
+  });
+
+  it("set&get nfs properties", async function () {
+    const posixProperties: FilePosixProperties = {
+      owner: "123",
+      group: "654",
+      fileMode: {
```

Comment:
Is there any other way to create the file mode, like from "0777"?
This is .net datalakeGen2 SDK sample: https://github.com/Azure/azure-sdk-for-net/blob/229349bb816ef982dafc62a32d550832b2670023/sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs#L502